src/features/codeLensProvider.ts

```diff
@@ -139,4 +141,41 @@ export default class OmniSharpCodeLensProvider extends AbstractProvider implemen
                 { title: "debug test", command: 'dotnet.test.debug', arguments: [testFeature.Data, fileName, testFrameworkName] }));
         }
     }
+
+    private _updateCodeLensForTestClass(bucket: vscode.CodeLens[], fileName: string, node: protocol.Node) {
+        //if the class doesnot contain any method then return
+        if (!node.ChildNodes.find(value => (value.Kind == "MethodDeclaration"))) {
+            return;
+        }
+
+        let testMethodsInClass = new Array<string>();
+        let testFrameworkName: string = null;
+        for (let child of node.ChildNodes) {
+            if (child.Kind == "MethodDeclaration") {
+                let testFeature = child.Features.find(value => (value.Name == 'XunitTestMethod' || value.Name == 'NUnitTestMethod' || value.Name == 'MSTestMethod'));
+                if (testFeature) {
+                    // this test method has a test feature
+                    if (testFrameworkName == null) {
+                        testFrameworkName = 'xunit';
```

Comment:
Is it possible for someone to have tests for different frameworks in the same class? What happens if they do?

Reply:
I don't think so. The code currently assumes that every test method in the class has the same frameworkname as the first testmethod in the class and the test request also contains only one field `TestFrameworkName` (https://github.com/akshita31/omnisharp-vscode/blob/504e97edf968229e9f14f8603b7ebeaf5b764ade/src/omnisharp/protocol.ts#L588) for a group of methods. Do you think this needs to change ?

Reply:
It's probably fine then. Thanks for checking.